src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -1367,14 +1369,15 @@ VecSimQueryResult *HierarchicalNSW<dist_t>::searchRangeBottomLayer_WithTimeout(
         dynamic_range = radius; // to ensure that dyn_range >= radius.
     }
 
+    dist_t dynamic_range_search_boundaries = dynamic_range * (1.0 + epsilon);
     candidate_set.emplace(-ep_dist, ep_id);
     this->visited_nodes_handler->tagNode(ep_id, visited_tag);
 
     while (!candidate_set.empty()) {
         std::pair<dist_t, tableint> curr_el_pair = candidate_set.top();
         // If the best candidate is outside the dynamic range in more than epsilon (relatively) - we
         // finish the search.
-        if ((-curr_el_pair.first) > dynamic_range * (1.0f + epsilon)) {
+        if ((-curr_el_pair.first) > dynamic_range_search_boundaries) {
             break;
```

Comment:
coverage says you don't get here. Isn't it strange that no test (even the flow) does not end here?

Reply:
We do not account for line of code that are covered only in flow tests (or in benchmark tests)... I believe that it is covered under flow tests (otherwise we would get perfect recall). Anyway, I adjusted the unit test to cover this case as well.